src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php

```diff
@@ -105,9 +105,9 @@ public function finishView(FormView $view, FormInterface $form, array $options)
             $tokenId = $options['csrf_token_id'] ?: ($form->getName() ?: get_class($form->getConfig()->getType()->getInnerType()));
             $data = (string) $options['csrf_token_manager']->getToken($tokenId);
 
-            $csrfForm = $factory->createNamed($options['csrf_field_name'], 'Symfony\Component\Form\Extension\Core\Type\HiddenType', $data, array(
+            $csrfForm = $factory->createNamed($options['csrf_field_name'], $options['csrf_field_type'], $data, array_replace(array(
```

Comment:
While I can understand that there may be use case where you need to (want to) customise the CSRF field's options, I do not see a need to change the type being used. Without a really good use case I am against making it configurable.